spotify.go

```diff
@@ -122,6 +139,25 @@ type Image struct {
 	URL string `json:"url"`
 }
 
+// UnmarshalJSON unmarshals the image data regardless of numeric values being integers or floats.
+func (i *Image) UnmarshalJSON(data []byte) error {
+	var v struct {
+		Height float64 `json:"height"`
+		Width  float64 `json:"width"`
+		URL    string  `json:"url"`
+	}
+
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	i.Height = int(v.Height)
```

Comment:
I wonder if longterm it makes sense for us to return these fields as floats rather than casting them back to ints. It'd be a breaking change though so perhaps worth bundling with some other fixes.

Reply:
I kept it int for now, but removed the unmarshallers in favour of a custom type (more below)